### PR TITLE
paste: fix multiple bugs

### PR DESCRIPTION
**Pull request by andrewliebenow**
- Implement handling of "\0" delimiter
- Process data in bytes to allow handling of non-UTF-8 data
- Forbid unescaped trailing backslash delimiter
- Fix handling of repeated input files
- In "-s" mode, reset delimiters iterator after each file

### CODE REVIEW

File: text/paste.rs [hunk 1]
```diff
@@ -8,53 +8,85 @@
 //
 // TODO:
 // - stdin ("-")
-// -- Probably fixed
+// -- Fixed
 // - fix:  empty-string delimiters \0
-// -- Probably fixed
+// -- Fixed
 // - improve:  don't open all files at once in --serial mode
```
Comment (jgarzik) [inline text/paste.rs:11]: If it's fixed, delete the TODO entry entirely.  No need add 'Fixed' annotation.

The history is recorded in git (and github, to a lesser extent).

File: text/paste.rs [hunk 2]
```diff
@@ -8,53 +8,85 @@
...
-    /// Concatenate all of the lines from each input file into one line of output per file, in command line order.
+    /// Concatenate all of the lines from each input file into one line of output per file, in command line order
     #[arg(short, long)]
     serial: bool,
 
     /// Delimiter list
-    #[arg(short, long)]
+    // Other implementations use "delimiters"
+    #[arg(alias = "delimiters", short, long)]
     delims: Option<String>,
```
Comment (jgarzik) [inline text/paste.rs]: In terms of long options,
* POSIX does not specify, so posixutils has complete freedom of choice
* Default is to be compatible with in-the-field long options (GNU utils, typically)
* "delims" is our own invention presumably, and by the above-stated logic, should be changed to `delimiters`
* conclusion:  long="delimiters" seems like a better choice, WDYT?

Comment (andrewliebenow) [reply text/paste.rs]: Agreed and changed

File: text/paste.rs (hunk 1 above)
Comment (andrewliebenow) [reply text/paste.rs:11]: Done

**Review by jgarzik — APPROVED** [review]

**Pull request merged**